hls-plugin-api/src/Ide/Types.hs

```diff
@@ -776,7 +776,7 @@ instance Semigroup (PluginNotificationHandlers a) where
 instance Monoid (PluginNotificationHandlers a) where
   mempty = PluginNotificationHandlers mempty
 
-type PluginMethodHandler a m = a -> PluginId -> MessageParams m -> LspM Config (Either ResponseError (MessageResult m))
+type PluginMethodHandler a m = a -> PluginId -> MessageParams m -> LspM Config (Either PluginError (MessageResult m))
```

Comment:
this could definitely end in `ExceptT PluginError (LspM Config) (MessageResult m)`, I think